bin/drone-chef-supermarket

```diff
@@ -1,28 +1,63 @@
 #!/usr/bin/env ruby
 
+require "gli"
+
 $LOAD_PATH.push File.expand_path("../../lib", __FILE__)
 
 Signal.trap("INT") do
   abort
 end
 
 begin
-  require "drone"
   require "drone/supermarket"
 rescue LoadError
   require "rubygems"
-  require "drone"
   require "drone/supermarket"
 end
 
-ARGV.delete "--"
+include GLI::App
 
-begin
-  Drone::Plugin.new(ARGV.first) do |payload|
-    Drone::Supermarket.new(
-      payload
-    ).execute!
+def load_env(options, key)
+  if options[key.downcase.to_sym].nil? && !ENV["PLUGIN_#{key.upcase}"].nil?
+    options[key.downcase.to_sym] = ENV["PLUGIN_#{key.upcase}"]
+    options[key] = ENV["PLUGIN_#{key.upcase}"]
   end
-rescue Interrupt
-  abort "Quitting..."
 end
+
+def load_secrets(options, key)
+  if options[key.downcase.to_sym].nil? &&
+     !ENV["SUPERMARKET_#{key.upcase}"].nil?
+    options[key.downcase.to_sym] = ENV["SUPERMARKET_#{key.upcase}"]
+    options[key] = ENV["SUPERMARKET_#{key.upcase}"]
+  end
+end
+
+program_desc "Drone Supermarket Plugin"
+
+command :upload do |c|
+  c.flag [:u, :user]
+  c.flag [:k, :private_key], mask: true
+  c.flag [:s, :server]
+  c.flag [:v, :ssl_verify]
```

Comment:
I'm having a hard time to find an example of this.  Looking at docs it says:

> It can take `:type`, which is a class for type conversion

But it seems that only takes 1 class? ... and there is no boolean class in Ruby as far as I know ... there is `TrueClass` and `FalseClass`.  So not sure how I would do it.